### ISSUE TITLE
Cryptonight hash self-test failed

### ISSUE THREAD

**Issue by meta0102** [fireice-uk/xmr-stak#1938]
if i try to start the new 2.5 version i get always the following error: (the old 2.4.x version worked fine)
```
[2018-10-15 10:46:05] : Cryptonight hash self-test failed. This might be caused by bad compiler optimizations.
[2018-10-15 10:46:05] : Cryptonight hash self-test failed. This might be caused by bad compiler optimizations.
[2018-10-15 10:46:05] : Self test not passed!
```

```
Distributor ID: Ubuntu
Description:    Ubuntu 17.04
Release:        17.04
Codename:       zesty

```

Please provide as much as possible information to reproduce the issue.

# Basic information
  - Type of the CPU - **Intel(R) Xeon(R) CPU E5-2670 v2 @ 2.50GHz**

# Issue with the execution
  - Do you compiled the miner by our own? **Yes**
  ```
Version: xmr-stak/2.5.0/9012512/master/lin/cpu/0
  ```
`cmake -DCUDA_ENABLE=OFF -DOpenCL_ENABLE=OFF ..`

**Comment by psychocrypt** [comment]
Please test the miner without any changes. Which currency do you try to mine?

**Comment by meta0102** [comment]
monero - for the fork
i've not changed anything. default config. only CPU is used

**Comment by meta0102** [comment]
fixed after updating ubuntu to 18.04
